File: Backlogr.Api/Models/Entities/Review.cs
namespace Backlogr.Api.Models.Entities;

public sealed class Review
{
    public Guid ReviewId { get; set; }

    public Guid UserId { get; set; }

    public Guid GameId { get; set; }

    public string Text { get; set; } = string.Empty;

    public bool HasSpoilers { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ApplicationUser User { get; set; } = null!;

    public Game Game { get; set; } = null!;

    public ICollection<ReviewLike> ReviewLikes { get; set; } = new List<ReviewLike>();

    public ICollection<ReviewComment> ReviewComments { get; set; } = new List<ReviewComment>();
}

File: Backlogr.Web/types/review.ts
// /types/review.ts

export interface CreateReviewRequestDto {
  gameId: string
  text: string
  hasSpoilers: boolean
}

export interface UpdateReviewRequestDto {
  text: string
  hasSpoilers: boolean
}

export interface CreateReviewCommentRequestDto {
  text: string
}

export interface ReviewResponseDto {
  reviewId: string
  userId: string
  userName: string
  displayName: string
  gameId: string
  gameTitle: string
  text: string
  hasSpoilers: boolean
  createdAt: string
  updatedAt: string
}

export interface ReviewCommentResponseDto {
  reviewCommentId: string
  reviewId: string
  userId: string
  userName: string
  displayName: string
  avatarUrl?: string | null
  text: string
  createdAt: string
  isOwner: boolean
}
